tests/readPreference/readpreference-set_state-001.phpt

```diff
@@ -48,31 +48,31 @@ foreach ($tests as $fields) {
 
 %r\\?%rMongoDB\Driver\ReadPreference::__set_state(array(
    'mode' => 'secondary',
-   'tags' => 
+   'tags' =>%w
   array (
-    0 => 
-    %Sarray(
+    0 =>%w
+    %r\(object\)? %rarray(
        'dc' => 'ny',
-    %S),
+    %r\)?%r),
   ),
 ))
 
 %r\\?%rMongoDB\Driver\ReadPreference::__set_state(array(
    'mode' => 'secondary',
-   'tags' => 
+   'tags' =>%w
   array (
-    0 => 
-    %Sarray(
+    0 =>%w
+    %r\(object\)? %rarray(
        'dc' => 'ny',
-    %S),
-    1 => 
-    %Sarray(
+    %r\)?%r),
+    1 =>%w
+    %r\(object\)? %rarray(
        'dc' => 'sf',
        'use' => 'reporting',
-    %S),
-    2 => 
-    %Sarray(
-    %S),
+    %r\)?%r),
+    2 =>%w
+    %r\(object\)? %rarray(
+    %r\)?%r),
```

Comment:
Why would we expect _another_ optional `)` character here? I understand an optional `(object)` cast in the output, but there is already an expected `)` to balance `array(` on the previous line. And there are other closing parens _below_ that balance out the `array(` instances up top.

I don't think this has anything to do with your previous related comments:

 * https://github.com/mongodb/mongo-php-driver/pull/1450#discussion_r1281459424
 * https://github.com/mongodb/mongo-php-driver/pull/1450#discussion_r1281459748

I see you have `%r\(object\)? %r` above, and the closing paren there is optional. Does that mean that PHP sometimes emits `(object array(...))`? That doesn't make sense as it wouldn't be valid syntax.

In fact, the `%r\(object\)? %r` pattern would only match `(object ` or `(object) `. In that case, it seems like `(object)` is _always_ present (it's the only valid syntax). So perhaps we should expect `(object)` for PHP 7.4+.

I tested this in https://3v4l.org/MaXQ3 and `(object)` is always included in PHP 7.3+. Previous versions would emit `stdClass::__set_state(array(...))`, which explains the extra closing paren, which is no longer applicable.

And PHP 8.2+ differs slightly by emitting a backslash before class names to make them fully-qualified. This is already accounted for with the `%r\\?%r` pattern earlier in the `EXPECTF` block.

----

Note: this applies to other `EXPECTF` sections with `__set_state()` output.

Reply:
Good observation, I believe I originally added this in error and then it spread via the usual way of copy pasta.